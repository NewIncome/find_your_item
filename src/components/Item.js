import React from 'react';
import PropTypes from 'prop-types';

const Item = props => {
  const { item, link } = props;

  return (
    <div className="Items-info">
      <h3 className="Items-name">{item.name}</h3>
      <img
        className="Items-image"
        src={link}
        alt={item.name}
      />
      <p className="Items-description">{item.description}</p>
      {item.id
        ? (
          <div>
            <p>{item.id}</p>
            <div>
              <p>{item.created_at}</p>
              <p>{item.updated_at}</p>
            </div>
          </div>
        )
        : ''}
    </div>
  );
};

Item.defaultProps = {
  link: '',
};

Item.propTypes = {
  item: PropTypes.objectOf(PropTypes.any).isRequired,
  link: PropTypes.string,
};

export default Item;
