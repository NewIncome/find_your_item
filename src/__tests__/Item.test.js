import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import Item from '../components/Item';

const item = {
  id: 1,
  name: 'item name',
  description: 'item description',
  created_at: '2021-12-02',
  updated_at: '2021-12-02',
}

describe('Item component', () => {
  test('is displayed correctly', () => {
    const tree = renderer.create(<Item item={item} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  test("has a 'price' prop", () => {
    const tree = renderer.create(<Item item={item} price={135} />).root;
    expect(tree.findByType(Item).props.price).toBe(135);
  });

  test("has an 'items' prop", () => {
    const tree = renderer.create(<Item item={item} />).root;
    expect(tree.findByType(Item).props.item).toBe(item);
  });

  test('has an h3 element', () => {
    const tree = renderer.create(<Item item={item} pg='' />).root;
    expect(tree.findByProps({ className: 'itm-name' }).children).toEqual([`${item.created_at} `]);
  });

  test('has an p element', () => {
    const tree = renderer.create(<Item item={item} />).root;
    expect(tree.findByProps({ className: 'stars' }).children).toEqual(['★★★★☆']);
  });
});
