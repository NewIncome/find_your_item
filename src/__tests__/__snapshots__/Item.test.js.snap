// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item component is displayed correctly 1`] = `
<div
  className="Items-info"
>
  <div
    className="image-frame"
  >
    <img
      alt="item name"
      className="Items-image"
    />
    <div
      className="img-info"
    >
      <div
        className="dtl-left  dtls"
      >
        <p
          className="stars"
        >
          ★★★★☆
        </p>
        <p
          className="itm-name"
        >
          2021-12-02 
        </p>
      </div>
      <div
        className="dtl-right dtls"
      >
        <p
          className="d-r-u"
        >
          $
          1000
        </p>
        <p
          className="d-r-d"
        >
          Price
        </p>
      </div>
    </div>
  </div>
  <div
    className="image-desc"
  >
    <p
      className="desc-title"
    >
      About this item
    </p>
    <p
      className="desc-description"
    >
      item description
    </p>
  </div>
</div>
`;
